service/src/main/resources/db/changesets/20220301_datasettable.yaml

```diff
@@ -0,0 +1,40 @@
+databaseChangeLog:
+  - changeSet:
+      id: dataset_table
+      author: sehsan
+      changes:
+        - createTable:
+            tableName: dataset
+            columns:
+              - column:
+                  name: id
+                  type: ${uuid_type}
+                  defaultValueComputed: ${uuid_function}
+                  constraints:
+                    primaryKey: true
+                    nullable: false
```

Comment:
All the operation are done through this `uuid` as a primary key. Right now it is possible to have multiple duplicate `dataset_id` and `storage_system` rows. Is this desirable? Also, earlier we discussed using a composite primary key instead (but not a string). At the UI layer, will we use these `uuid`s? Or will there be a translation from `tdr_<uuid>` to this `uuid`? I think it would be convenient if we encode some additional information in `uuid`s (probably not at the database layer), that way users at least know they're using the right `uuid` for their systems, and we detect some simple things with this too ("oh, you're trying to add a TDR id to the wrong storage system"). GitHub for example does this for personal access tokens: they're opaque blobs that have a `ghp_` prefix, which enables them to scan for secrets accidentally being committed in repositories.

Reply:
The unique constraint will ensure that the combination of `dataset_id` and `storage_system` will be unique (I added some unit tests to check this behavior). 

I was thinking as we eventually support different storage systems, it might confuse users to see ids in multiple different formats. For instance, what would the id for an "external" dataset look like (where we just provide a link to the dataset)? We could still validate that the dataset_id is in a valid format for the storage system when a user tries to create a catalog entry. What do you think?